bootstrap/kubeadm/types/utils.go

```diff
@@ -65,46 +82,49 @@ func KubeVersionToKubeadmAPIGroupVersion(v semver.Version) (schema.GroupVersion,
 	case version.Compare(v, v1beta3KubeadmVersion, version.WithoutPreReleases()) < 0:
 		// NOTE: All the Kubernetes version >= v1.15 and < v1.22 should use the kubeadm API version v1beta2
 		return upstreamv1beta2.GroupVersion, nil
-	default:
-		// NOTE: All the Kubernetes version greater or equal to v1.22 should use the kubeadm API version v1beta3.
-		// Also future Kubernetes versions (not yet released at the time of writing this code) are going to use v1beta3,
-		// no matter if kubeadm API versions newer than v1beta3 could be introduced by those release.
-		// This is acceptable because v1beta3 will be supported by kubeadm until the deprecation cycle completes
-		// (9 months minimum after the deprecation date, not yet announced now); this gives Cluster API project time to
-		// introduce support for newer releases without blocking users to deploy newer version of Kubernetes.
+	case version.Compare(v, v1beta4KubeadmVersion, version.WithoutPreReleases()) < 0:
+		// NOTE: All the Kubernetes version >= v1.22 and < v1.31 should use the kubeadm API version v1beta3
 		return upstreamv1beta3.GroupVersion, nil
+	default:
+		// NOTE: All the Kubernetes version greater or equal to v1.31 should use the kubeadm API version v1beta4.
+		// Also future Kubernetes versions (not yet released at the time of writing this code) are going to use v1beta4,
```

Comment:
Not a problem of this PR. I wonder how true this statement is actually. 

If v1.31 would use v1beta4 and we wouldn't have merged this PR, would everything work if we continue to just use v1beta3?